app/controllers/npq_separation/admin/finance/statements/adjustments_controller.rb

```diff
@@ -20,14 +23,19 @@ def create
 
   def index
     @add_another_form = Admin::Adjustments::AddAnotherAdjustmentForm.new(add_another_params.merge(statement: @statement))
-    @create_adjustment_form = Admin::Adjustments::CreateAdjustmentForm.new(session:, statement: @statement)
+    @create_adjustment_form = Admin::Adjustments::CreateAdjustmentForm.new(created_adjustment_ids: session[:created_adjustment_ids], statement: @statement)
     @adjustments = @create_adjustment_form.adjustments
   end
 
   def add_another
     index
     if @add_another_form.valid?
-      redirect_to @add_another_form.redirect_to
+      if @add_another_form.adding_another_adjustment?
+        redirect_to new_npq_separation_admin_finance_statement_adjustment_path(@statement)
+      else
+        session[:created_adjustment_ids] = nil
+        redirect_to npq_separation_admin_finance_statement_path(@statement)
+      end
     else
       render :index, status: :unprocessable_entity
     end
```

Comment:
Super minor but this could be 'inverted' to remove a layer of nesting and simplify readability

```ruby
if form.invalid?
  render :index, status: :unprocessable_entity
elsif form.adding_another?
  # redirect to add another screen
else
  # redirect to statement page
end
```